### DISCUSSION TITLE
Connection Manager replacement issues

### DISCUSSION THREAD

**Discussion by JacoJordaan** [MarimerLLC/csla#3752] [Questions]
I have ported my Csla 5 to the latest Csla 8 code and are running into an issue with transaction handling in my DAL.
In Csla 5 the connection manager was used it it worked fine. In Csla 8 I am using the suggested option as per #3287

I am receiving 2 types of errors
1. Trying to enlist in ditributed transaction related errors
2. Connection pool errors: `The timeout period elapsed prior to obtaining a connection from the pool.  This may have occurred because all pooled connections were in use and max pool size was reached.`

I am using ADO.NET code in my DAL. I am also using the transaction scope attributes in the data portal call in the business object e.g. [Transactional(TransactionalTypes.TransactionScope)]
I am using only one connection string

This is my connection class:
````
public class QSDbConnection : IQSDbConnection
{   
    private readonly ILogger<QSDbConnection> _logger;
    public QSDbConnection(ILogger<QSDbConnection> logger)
    {
        _logger = logger;
        _logger.LogInformation("Init QSDbConnection");            
    }

    public SqlConnection GetConnection(string connectionString)
    {        
        var conn = new SqlConnection(connectionString);   
        conn.Open();
        return conn;
    }

    public IDbConnection GetDapperConnection(string connectionString)
    {        
        return new SqlConnection(connectionString);
    }
}
````

Calls to the connection is wrapped in using statements e.g.:
````
using (var ctx = conn.GetConnection("QSDb"))
{
    using (var cm = ctx.CreateCommand())
    {        
        cm.CommandText = RequiredPartBase_FetchSql;
        cm.CommandType = System.Data.CommandType.Text;
        cm.Parameters.AddWithValue("@ID", id);
		....
````

A few questions comes to mind:
- Has anyone experience the same issue and was able to resolve this?
- Is the using of the transaction scope attributes in the business object still a valid way of handling transactions?

**Comment by Inictus** [comment]
Yeah I had this problem when upgrading from CSLA 4 to 7 (haven't tried 8 quite yet). I had to piece together a few different posts and a lot of experimentation.

First, Rocky talks about it here: https://stackoverflow.com/questions/72295421/upgrade-to-csla-6-connectionmanager-problem
And
https://github.com/MarimerLLC/csla/discussions/2979

These helped me alot. 

The transaction scope attributes still work just fine when placed on the portal methods on the business object.
